Tests/helper.py

```diff
@@ -86,28 +87,32 @@ def assert_image(im, mode, size, msg=None):
         )
 
 
-def assert_image_equal(a, b, msg=None):
+def assert_image_equal(a: Image.Image, b: Image.Image, msg: str | None = None) -> None:
     assert a.mode == b.mode, msg or f"got mode {repr(a.mode)}, expected {repr(b.mode)}"
     assert a.size == b.size, msg or f"got size {repr(a.size)}, expected {repr(b.size)}"
     if a.tobytes() != b.tobytes():
-        if HAS_UPLOADER:
+        if uploader:
```

Comment:
This `if` is probably no longer needed since it is now checked in the `upload` function.

Reply:
If this condition was removed, then the `logger.error` on line 97 would always run, and similarly, the `logger.exception` on line 136 would always run.

Reply:
Perhaps it shouldn't run if there is no url returned?
If `uploader = "show"`, that logger message is probably also undesirable.
```python
url = upload(a, b)
if url:
    logger.error("URL for test images: %s", url)
```

Reply:
Ok, I've pushed https://github.com/python-pillow/Pillow/pull/7733/commits/f7701e6596c9a2e7c9ed886621f01b3861d70990 for this.